src/curve_intersections.rs

```diff
@@ -0,0 +1,1377 @@
+        let d1 = signed_distance_from_ray_to_point(l, self.p1);
+        let d2 = signed_distance_from_ray_to_point(l, self.p2);
+        let d3 = signed_distance_from_ray_to_point(l, self.end());
+
+        let p0 = Point::new(0.0, d0);
+        let p1 = Point::new(1.0 / 3.0, d1);
+        let p2 = Point::new(2.0 / 3.0, d2);
+        let p3 = Point::new(1.0, d3);
+        // Compute the vertical signed distance of p1 and p2 from [p0, p3].
+        let dist1 = d1 - (2.0 * d0 + d3) / 3.0;
+        let dist2 = d2 - (d0 + 2.0 * d3) / 3.0;
+
+        // Compute the hull assuming p1 is on top - we'll switch later if needed.
+        let mut hull = if dist1 * dist2 < 0.0 {
+            // p1 and p2 lie on opposite sides of [p0, p3], so the hull is a quadrilateral:
+            (vec![p0, p1, p3], vec![p0, p2, p3])
+        } else {
+            // p1 and p2 lie on the same side of [p0, p3]. The hull can be a triangle or a
+            // quadrilateral, and [p0, p3] is part of the hull. The hull is a triangle if the vertical
+            // distance of one of the middle points p1, p2 is <= half the vertical distance of the
+            // other middle point.
+            let dist1 = dist1.abs();
+            let dist2 = dist2.abs();
+            if dist1 >= 2.0 * dist2 {
+                (vec![p0, p1, p3], vec![p0, p3])
+            } else if dist2 >= 2.0 * dist1 {
+                (vec![p0, p2, p3], vec![p0, p3])
+            } else {
+                (vec![p0, p1, p2, p3], vec![p0, p3])
+            }
+        };
+
+        // Flip the hull if needed:
+        if dist1 < 0.0 || (dist1 == 0.0 && dist2 < 0.0) {
+            hull = (hull.1, hull.0);
+        }
+
+        hull
+    }
+}
+
+bitflags::bitflags! {
+    pub struct CurveIntersectionFlags: u32 {
+        const NONE                                  = 0b00000000;
+        const KEEP_CURVE1_T0_INTERSECTION           = 0b00000001;
+        const KEEP_CURVE1_T1_INTERSECTION           = 0b00000010;
+        const KEEP_CURVE2_T0_INTERSECTION           = 0b00000100;
+        const KEEP_CURVE2_T1_INTERSECTION           = 0b00001000;
+        const KEEP_CURVE1_ENDPOINT_INTERSECTIONS    = Self::KEEP_CURVE1_T0_INTERSECTION.bits
+                                                    | Self::KEEP_CURVE1_T1_INTERSECTION.bits;
+        const KEEP_CURVE2_ENDPOINT_INTERSECTIONS    = Self::KEEP_CURVE2_T0_INTERSECTION.bits
+                                                    | Self::KEEP_CURVE2_T1_INTERSECTION.bits;
+        const KEEP_ALL_ENDPOINT_INTERSECTIONS       = Self::KEEP_CURVE1_ENDPOINT_INTERSECTIONS.bits
+                                                    | Self::KEEP_CURVE2_ENDPOINT_INTERSECTIONS.bits;
+        const KEEP_DUPLICATE_INTERSECTIONS          = 0b00010000;
+    }
+}
+
+/// Compute the intersections between two Bézier curves
+pub fn curve_curve_intersections<T: ParamCurveBezierClipping, U: ParamCurveBezierClipping>(
+    curve1: &T,
+    curve2: &U,
+    flags: CurveIntersectionFlags,
```

Comment:
I think we have a problem here, the fn is publicly exported, but the flags type is not.

Reply:
`CurveIntersectionFlags` is public. Or am I missing something?

See curve_intersections.rs line 259: `pub struct CurveIntersectionFlags`